qcodes/utils/oscilloscope.py

```diff
@@ -0,0 +1,252 @@
+import traceback
+import numpy as np
+import multiprocessing as mp
+import logging
+import time
+
+import pyqtgraph as pg
+import pyqtgraph.multiprocess as pgmp
+from pyqtgraph import QtGui
+from pyqtgraph.multiprocess.remoteproxy import ClosedError
+
+
+logger = logging.getLogger(__name__)
+
+# https://stackoverflow.com/questions/17103698/plotting-large-arrays-in-pyqtgraph?rq=1
+
+
+class Oscilloscope():
+    """Create an oscilloscope GUI that displays acquisition traces
+
+    Example code:
+
+    ```
+    oscilloscope = Oscilloscope(channels=['chA', 'chB', 'chC', 'chD'])
+    oscilloscope.start_process()
+
+    # Define settings
+    oscilloscope.ylim = (-0.8, 1)
+    oscilloscope.channels_settings['chB']['scale'] = 10
+    oscilloscope.channels_settings['chC']['scale'] = 50
+    oscilloscope.channels_settings['chD']['scale'] = 50
+    oscilloscope.update_settings()
+
+    # Register oscilloscope update event with acquisition controller
+    triggered_controller.buffer_actions = [oscilloscope.update_array]
+    """
+    def __init__(
+            self,
+            channels: list,
+            max_points=200000,
+            channels_settings=None,
+            figsize=(1200, 350),
+            sample_rate=200e3,
```

Comment:
can we ask sample rate from digitizer directly?

Reply:
ah, wait, it's in qcodes, different layer of instrument code... I guess not then...